contracts/SubdomainRegistrar.sol

```diff
@@ -60,12 +64,24 @@ contract SubdomainRegistrar is RegistrarInterface {
         _;
     }
 
+    modifier not_stopped() {
+        require(!stopped);
+        _;
+    }
+
+    modifier registrar_owner_only() {
+        require(msg.sender == registrarOwner);
+        _;
+    }
+
     event TransferAddressSet(bytes32 indexed label, address addr);
     event DomainUpgraded(bytes32 indexed label, string name);
+    event DomainMigrated(bytes32 indexed label);
```

Comment:
Instead of an event just for migration, maybe make an event for transfer, which can trigger any time the domain's transferred out?